### PR TITLE
Rename `TimeZone::timestamp*` to `TimeZone::and_timestamp*`

### PR DESCRIPTION
**Pull request by pitdicker**
We have the pattern to use the name `and_` for methods that add information to convert to a more complete type.
`NaiveDate` had `and_time` and `and_hms*`, `NaiveDateTime` has `and_local_timezone`.

This makes the methods on `TimeZone` consistent. `TimeZone::timestamp*` is renamed to `TimeZone::and_timestamp*` and `TimeZone::with_ymd_and_hms` to `TimeZone::and_ymd_and_hms*`.

### CODE REVIEW

**Comment by djc** [comment]
So we may have this `and_()` pattern but I feel like `with_()` is much more common in the Rust ecosystem. Maybe we should be moving in the other direction instead? (I think that was the idea behind `with_ymd_and_hms()`, at least.)

Can we do some part of this on main?

**Comment by pitdicker** [comment]
Fine by me. Then we should probably rename the current `with_` methods such as `with_year` to something like `replace_year`? That matches time-rs.

**Comment by djc** [comment]
How are the semantics for current `with_()` methods different from those for current `and_()` methods, in your opinion?

**Comment by pitdicker** [comment]
Currently the `with_()` methods return the same type and replace some part of the date, time or offset.
The `and_()` methods add more information to turn it into another type.

**Comment by pitdicker** [comment]
Other methods that start with `and` are `NaiveDate::{and_time, and_hms*}`, `NaiveDateTime::{and_local_timezone, and_utc}`.

And one more that starts with `with` is `DateTime::with_timezone`.

**Comment by pitdicker** [comment]
I'll collect some info in an issue.

**Comment by pitdicker** [comment]
Added commits to rename all methods from #1527 except `TimeZone::{from_local_datetime, from_utc_datetime}`.

I am not 100% conviced with the new names for `NaiveDateTime::{and_local_timezone, and_utc}`. Naming them `in_` does match natural language. And `assume_` seems like saying the user may not be sure for some reason.

Example code:
```rust
let pst = FixedOffset::east(8 * 60 * 60)?;
// in_timezone
let dt = NaiveDate::from_ymd(2018, 1, 11)?.at_hms(10, 5, 13)?.in_timezone(pst);
// assume_timezone
let dt = NaiveDate::from_ymd(2018, 1, 11)?.at_hms(10, 5, 13)?.assume_timezone(pst);

// in_utc
let timestamp = NaiveDate::from_ymd(1970, 1, 1)?.at_hms(0, 0, 0).in_utc().timestamp();
// assume_utc
let timestamp = NaiveDate::from_ymd(1970, 1, 1)?.at_hms(0, 0, 0).assume_utc().timestamp();
```

File: src/datetime/mod.rs
Comment (djc) [inline src/datetime/mod.rs]: This seems weird if a later commit is just going to change it again to `in_utc()`.

**Review by djc — APPROVED** [review]

File: src/datetime/mod.rs
Comment (pitdicker) [reply src/datetime/mod.rs]: I agree. This commit is merged on the main branch in #1530. I'll wait until main is merged into the 0.5.x branch again and then rebase this PR.

**Pull request merged**
